flink-core/src/main/java/org/apache/flink/api/common/JobSubmissionResult.java

```diff
@@ -56,10 +56,6 @@ public boolean isJobExecutionResult() {
 	 * @throws ClassCastException if this is not a JobExecutionResult
 	 */
 	public JobExecutionResult getJobExecutionResult() {
-		if (isJobExecutionResult()) {
-			return (JobExecutionResult) this;
-		} else {
-			throw new ClassCastException("This JobSubmissionResult is not a JobExecutionResult.");
-		}
+		throw new ClassCastException("This JobSubmissionResult is not a JobExecutionResult.");
```

Comment:
In fact this infer a bad inheritance...

Reply:
I agree, but this is `public` unfortunately.